src/scatter_buildblock/ScatterEstimationByBin.cxx

```diff
@@ -11,447 +11,1274 @@
+                 ++segment_num)
+            {
+                SegmentByView<float> segment_by_view =
+                        this->norm_projdata_sptr->get_segment_by_view(segment_num);
+
+                in_place_apply_function(segment_by_view,
+                                        inverse_object);
+
+                if (!(inv_norm_projdata_sptr->set_segment(segment_by_view) == Succeeded::yes))
+                {
+                    warning("Error set_segment %d\n", segment_num);
+                    return Succeeded::no;
+                }
+            }
+
+            if (this->export_2d_projdata)
+            {
+                size_t lastindex = this->norm_projdata_filename.find_last_of(".");
+                std::string rawname = this->norm_projdata_filename.substr(0, lastindex);
+                std::string out_filename = rawname + "_2d.hs";
+
+                this->norm_projdata_2d_sptr.reset(new ProjDataInterfile(this->input_projdata_2d_sptr->get_exam_info_sptr(),
+                                                                        this->proj_data_info_2d_sptr,
+                                                                        out_filename,
+                                                                        std::ios::in | std::ios::out | std::ios::trunc));
+            }
+            else
+                this->norm_projdata_2d_sptr.reset(new ProjDataInMemory(this->input_projdata_2d_sptr->get_exam_info_sptr(),
+                                                                       this->proj_data_info_2d_sptr));
+
+            SSRB(*this->norm_projdata_2d_sptr,
+                 *inv_norm_projdata_sptr,false);
+
+            for (int segment_num = this->norm_projdata_2d_sptr->get_min_segment_num();
+                 segment_num <= this->norm_projdata_2d_sptr->get_max_segment_num();
+                 ++segment_num)
+            {
+                SegmentByView<float> segment_by_view =
+                        this->norm_projdata_2d_sptr->get_segment_by_view(segment_num);
+
+                in_place_apply_function(segment_by_view,
+                                        inverse_object);
+
+                if (!(this->norm_projdata_2d_sptr->set_segment(segment_by_view) == Succeeded::yes))
+                {
+                    warning("Error set_segment %d\n", segment_num);
+                    return Succeeded::no;
+                }
+            }
+        }
+        else
+        {
+            error(boost::format("The normalisation projdata %1% are not 3D") % this->norm_projdata_filename);
+        }
+
+        normalisation_coeffs.reset(new BinNormalisationFromProjData(this->norm_projdata_2d_sptr));
+    }
+    //<- End Normalisation ProjData
+
+    this->multiplicative_data_2d_sptr.reset(
+                new ChainedBinNormalisation(attenuation_correction, normalisation_coeffs));
+
+    iterative_object->get_objective_function_sptr()->set_normalisation_sptr(multiplicative_data_2d_sptr);
```

Comment:
only for iterative case? analytic will need something. probably best to do this norm stuff before the recon_setup and move this in there.